test_truffle/staking_factory.test.js

```diff
@@ -0,0 +1,115 @@
+const { assertRevert } = require('../test_embark/helpers/assertThrow')
```

Comment:
I wrote this test directly with Truffle since we already decided to drop Embark in favor of it.